cas-server-core/src/main/java/org/jasig/cas/AbstractCentralAuthenticationService.java

```diff
@@ -150,6 +159,15 @@ protected final void doPublishEvent(final ApplicationEvent e) {
 
 
     /**
+     * Gets a synchronization handle.
+     * @param id String mutex id
+     * @return SimpleMutex
+     */
+    protected final SimpleMutex getMutex(final String id) {
```

Comment:
I don't remember exactly. Also, looking at this code again, the `SimpleMutex` class should be using a `WeakReference` anyway. As written, these object will keep them selves alive since they have a standard strong reference to their keys.